### PR TITLE
[IDP-1391] Validate and apply nullable on nested schema properties

### PR DESCRIPTION
**Pull request by heqianwang**
## Description of changes
Nested records where the parent class does not have a nullable property placed all property of the child class as nullable. This is seems to be an issue with Swashbuckle. As such, the fix is to parse the nested schema properties and match with the reflected class properties.

## Breaking changes
None

## Additional checks
Run system tests.

- [ ] Updated the documentation of the project to reflect the changes
- [x] Added new tests that cover the code changes

### CODE REVIEW

File: src/Workleap.Extensions.OpenAPI/RequiredType/ExtractRequiredAttributeFromNullableType.cs [hunk 1]
```diff
@@ -25,4 +28,32 @@ public void Apply(OpenApiSchema schema, SchemaFilterContext context)
             schema.Required.Add(property.Key);
         }
     }
+
+    private static void ApplyNullabilityOnNestedSchemaProperties(OpenApiSchema schema, SchemaFilterContext context)
```
Comment (PrincessMadMath) [inline src/Workleap.Extensions.OpenAPI/RequiredType/ExtractRequiredAttributeFromNullableType.cs:32]: Should it be recursive for nested of nested?

File: src/tests/WebApi.OpenAPI.SystemTest/openapi-v1.yaml [hunk 2]
```diff
@@ -290,22 +298,16 @@ components:
...
-        - requiredStringProperty
+        - examples
       type: object
       properties:
-        requiredIntProperty:
-          type: integer
-          format: int32
-        requiredStringProperty:
-          type: string
+        examples:
```
Comment (PrincessMadMath) [inline src/tests/WebApi.OpenAPI.SystemTest/openapi-v1.yaml:304]: We should also test 2 level of nesting

File: src/Workleap.Extensions.OpenAPI/RequiredType/ExtractRequiredAttributeFromNullableType.cs [hunk 3]
```diff
@@ -25,4 +28,32 @@ public void Apply(OpenApiSchema schema, SchemaFilterContext context)
...
+    {
+        var nullabilityInfoContext = new NullabilityInfoContext();
+        var contextProperties = context.Type.GetProperties();
+        foreach (var (name, property) in schema.Properties)
```
Comment (PrincessMadMath) [inline src/Workleap.Extensions.OpenAPI/RequiredType/ExtractRequiredAttributeFromNullableType.cs:36]: I am not familiar with the feature of NullabilityInfoContext is there a link we could add as comment in code that explain this concept?

**Review by PrincessMadMath — APPROVED** [review]

File: src/tests/WebApi.OpenAPI.SystemTest/openapi-v1.yaml (hunk 2 above)
Comment (heqianwang) [reply src/tests/WebApi.OpenAPI.SystemTest/openapi-v1.yaml:345]: yup added another layer

**Review by PrincessMadMath — APPROVED** [review]

**Pull request merged**